### PR TITLE
Add pinned icon to pinned block in situ

### PR DESCRIPTION
**Pull request by marjisound**
## What does this change?
Pinned post feature was implemented recently. This PR is adding pinned icon to the pinned block in situ (the pinned block that's among the other blocks)

Note: I had to add a margin-left to the video or image element in blocks because otherwise, the image would overlay the whole block and the icon would go underneath it. 

## Does this change need to be reproduced in dotcom-rendering ?

- [ ] No
- [x] Yes (please indicate your plans for DCR Implementation)

## Screenshots
| Before      | After      |
|-------------|------------|
| image | image |

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

**Comment by alinaboghiu** [comment]
- [ ] Just noticing some extra space above the timestamp on each post that this introduces. I don't think we want that. 
- [ ] The pin icon in the designs is as wide as the smaller timestamp, maybe that can be tweaked to match please

File: static/src/stylesheets/module/content-garnett/_live-blog.scss [hunk 1]
```diff
@@ -388,6 +388,29 @@ $block-padding-left: gs-span(1) + 4;
     padding: 3px 0 1px 5px;
 }
 
+.pinned-icon-situ {
+    white-space: nowrap;
+    text-overflow: ellipsis;
+    min-width: 32px;
```
Comment (joecowton1) [inline static/src/stylesheets/module/content-garnett/_live-blog.scss:394]: Could / should the `px` values be REM? I think that's how they are in the rest of Frontend.

File: article/app/views/liveblog/liveBlogBlocks.scala.html [hunk 2]
```diff
@@ -5,5 +5,5 @@
 
 
 @blocks.map { block =>
-    @views.html.liveblog.liveBlogBlock(block, article, timezone)
+    @views.html.liveblog.liveBlogBlock(block, article, timezone, false, block.attributes.pinned == true)
```
Comment (alinaboghiu) [inline article/app/views/liveblog/liveBlogBlocks.scala.html:8]: could we do this? 
```suggestion
    @views.html.liveblog.liveBlogBlock(block, article, timezone, false, block.attributes.pinned)
```

**Comment by marjisound** [comment]
@alinaboghiu >The pin icon in the designs is as wide as the smaller timestamp, maybe that can be tweaked to match please. 

I think this is because of the actual numbers. If you change the time to match the one in design (09:38), the icon will be as wide as the smaller timestamp 
image

But for the time 17:07 it would look like this
image

Similar to the measurements in design, the circle width and height are 32px and the font size is GuardianTextSans 12px.

File: article/app/views/liveblog/liveBlogBlocks.scala.html (hunk 2 above)
Comment (marjisound) [reply article/app/views/liveblog/liveBlogBlocks.scala.html:8]: OMG I don't know what I was thinking 😄

File: article/app/views/liveblog/liveBlogBlock.scala.html [hunk 3]
```diff
@@ -3,7 +3,7 @@
 @import model.liveblog.{LiveBlogDate, BodyBlock}
 @import views.BodyProcessor
 @import model.DotcomContentType
-@(block: BodyBlock, article: Article, timezone: DateTimeZone, isPinnedBlock: Boolean = false)(implicit request: RequestHeader, context: model.ApplicationContext)
+@(block: BodyBlock, article: Article, timezone: DateTimeZone, isPinnedBlock: Boolean = false, isPinnedBlockInSitu: Boolean = false)(implicit request: RequestHeader, context: model.ApplicationContext)
```
Comment (abeddow91) [inline article/app/views/liveblog/liveBlogBlock.scala.html:6]: I wonder if there's a better name for this. Its a tricky one to name for sure! Would `isOriginalPinnedBlock` be clearer? I'm not really sure so keen to hear thoughts. Happy to keep with what is here but it feels like the phrase `in situ` has thrown a couple of people already.

File: static/src/stylesheets/module/content-garnett/_live-blog.scss (hunk 1 above)
Comment (marjisound) [reply static/src/stylesheets/module/content-garnett/_live-blog.scss:394]: I was also doubtful about this. I noticed that in most places in frontend they are using `px` in scss but it seems when it's converted to css, the `px`s are also converted to `rem`. So in this case `32px` is rendered as `2rem`

File: article/app/views/liveblog/liveBlogBlock.scala.html (hunk 3 above)
Comment (marjisound) [reply article/app/views/liveblog/liveBlogBlock.scala.html:6]: Actually I myself didn't even know what it meant 😃  I'm happy with the name you suggested `isOriginalPinnedBlock`
Does anyone else have any suggestion for this naming? @alinaboghiu @joecowton1

Comment (joecowton1) [reply article/app/views/liveblog/liveBlogBlock.scala.html]: I think that new suggestion is probably good, it makes it easier to understand anyway :)

File: article/app/views/liveblog/liveBlogBlock.scala.html [hunk 4]
```diff
@@ -31,11 +31,16 @@
...
-        </a>
-    </p>
+    <div class="block-time">
+        <p class="published-time">
+            <a href="/@article.metadata.id?page=with:block-@block.id#block-@block.id" itemprop="url" class="block-time__link">
+            @views.html.liveblog.dateBlock(block.referenceDateForDisplay().map(LiveBlogDate(_, timezone)), !isPinnedBlock, true)
+            </a>
+        </p>
+        @if(isOriginalPinnedBlock){
+            <span>@fragments.inlineSvg("pin", "icon", List("inline-icon", "pinned-icon-situ", "pinned-block__icon"))</span>
```
Comment (abeddow91) [inline article/app/views/liveblog/liveBlogBlock.scala.html:41]:
```suggestion
            <span>@fragments.inlineSvg("pin", "icon", List("inline-icon", "pinned-icon-original", "pinned-block__icon"))</span>
```

Wonder if we should update the word situ here too to reflect the earlier change?

Comment (marjisound) [reply article/app/views/liveblog/liveBlogBlock.scala.html]: Thanks for spotting this Anna, just changed it :)

**Comment by alinaboghiu** [comment]
You are perfectly right re width @marjisound
- [ ] Adding another task here for fixing the padding above the pin icon as discussed

**Comment by marjisound** [comment]
> You are perfectly right re width @marjisound
> 
> * [x]  Adding another task here for fixing the padding above the pin icon as discussed

Thanks @alinaboghiu for pointing out the spacing issue. I made some small changes to fix that. What do you think?

**Review by alinaboghiu — APPROVED** [review]

**Comment by alinaboghiu** [comment]
Perfect thanks Marjan!

**Pull request merged**